hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -165,9 +166,15 @@ public ConfigurationSource getConf() {
    */
   public boolean hasEnoughSpace(DatanodeDetails datanodeDetails,
       long sizeRequired) {
-    SCMNodeMetric nodeMetric = nodeManager.getNodeStat(datanodeDetails);
-    return (nodeMetric != null) && (nodeMetric.get() != null)
-        && nodeMetric.get().getRemaining().hasResources(sizeRequired);
+    if (datanodeDetails instanceof DatanodeInfo) {
+      DatanodeInfo datanodeInfo = (DatanodeInfo) datanodeDetails;
+      for (StorageReportProto reportProto : datanodeInfo.getStorageReports()) {
+        if (reportProto.getRemaining() > sizeRequired) {
+          return true;
+        }
+      }
+    }
+    return false;
```

Comment:
I think we need to distinguish two "false" cases here, one is datanodeDetails is not a DatanodeInfo, another is this DN doesn't have enough space.  The latter case is normal, but the formal case indidates that current solution is not perfect.  It would be better if there is a WARN or INFO log so that we can find the issue quickly.

Reply:
@ChenSammi sure, we could add a WARN for the first 'false' case, if hit then it implies that we should find a better solution here.

Reply:
Added a precondition check instead of a LOG, since it is not directly related to the service provided here.